main-app/app/scripts/providers/generate-deck.js

```diff
@@ -0,0 +1,24 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.SnapCardGame.Deck')
+        .factory('GenerateDeck', ['ShuffleAndDeal', function (ShuffleAndDeal) {
+            var DeckGeneration = function () {
+                var deck = [];
+                this.makeDeck = function () {
+                    var suits = [1, 2, 3, 4],
+                        cardsPerSuit = 5,
+                        numberOfSuits = 4;
+                    for (var i = 0; i < numberOfSuits; i++) {
+                        for (var j = 0; j < cardsPerSuit; j++) {
+                            deck[i * cardsPerSuit + j] = suits[i];
```

Comment:
Don't use suits (see above), if you *must* number 1-4 (0-3 will be easier from a code point of view, and won't effect the player) you can use:
    deck[i * cardsPerSuit + j] = i + 1;
But 0-3 will be easier